### PR TITLE
Log handler and Loki application plugins

### PR DESCRIPTION
**Pull request by vyzigold**
This pull request adds the log handler and loki application plugins.
I'll add tests for it this or next week.

### CODE REVIEW

File: plugins/application/loki/main.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+	l.config = &LokiConfig {
+		Connection:  "",
+		BatchSize:   20,
+		MaxWaitTime: 100,
+	}
+	err := config.ParseConfig(bytes.NewReader(c), l.config)
+	if err != nil {
+		return err
+	}
+	fmt.Println(l.config.MaxWaitTime)
```
Comment (paramite) [inline plugins/application/loki/main.go]: I guess this should be deleted now :).

**Review by paramite — COMMENTED** [review]
Besides one nit this is perfectly fine. 

We just need to wait now to merge Paul's PR to my PR to Pauls PR (:D see [1] for explanation), because the Event transport interface gonna change a bit. 

Or you can rebase this PR to [1] and adapt it to the new interface.

[1] https://github.com/paramite/sg-core/pull/1

**Review by pleimer — CHANGES_REQUESTED** [review]
Some debug code left in, but otherwise LGTM. This will have to be re based on on infrawatch/sg-core master branch and be adapted to work with the new events bus. It would also be good to see some unit tests added, but they can be added in a different PR. I look forward to having this as an addition :)

Side note: don't worry about failing CI, it's not going to work until I merge the CI fixes branch

File: plugins/application/loki/main.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+	}
+}
+
+// ReceiveEvent ...
+func (l *Loki) ReceiveEvent(hName string, eType data.EventType, msg string) {
+	switch eType {
+	case data.LOG:
+		log, err := lib.CreateLokiLog(msg)
+		a, err := json.Marshal(log)
+		fmt.Println(string(a))
```
Comment (pleimer) [inline plugins/application/loki/main.go]: The print statement is the best debugging tool 🙂

File: plugins/application/loki/main.go [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+		if err == nil {
```
Comment (pleimer) [inline plugins/application/loki/main.go]: standard way:
```go
if err != nil {
  l.logger.Metadata(logging.Metadata{"plugin": "loki", "log": msg})
  l.logger.Info("failed to parse the data in event bus - disregarding")
  return
}
l.logChannel <- log
```
The linter (once it is fixed) will probably complain about it

File: plugins/transport/socket/main.go [hunk 4]
```diff
@@ -81,6 +81,7 @@ func (s *Socket) Run(ctx context.Context, w transport.WriteFn, done chan bool) {
 	go func() {
 		for {
 			n, err := pc.Read(msgBuffer)
+			msgBuffer[n] = 0
```
Comment (pleimer) [inline plugins/transport/socket/main.go:84]: Why did you need to do this?

Comment (vyzigold) [reply plugins/transport/socket/main.go:84]: I just tried without this line again. I'm getting `invalid character 'd' after top-level value` when I receive a message that is shorter than the previous one and try to unmarshal it. So I add a 0 into the buffer right after the received message. Maybe I'm not unmarshaling it in the handler correctly?

**Comment by vyzigold** [comment]
I resolved everything you pointed out. I have no idea, how I overlooked all the debugging statements :smile:. I'll close this PR, rebase to the current infrawatch sg-core master and create a new PR there.

File: plugins/transport/socket/main.go (hunk 4 above)
Comment (pleimer) [reply plugins/transport/socket/main.go:84]: oh wow I think you spotted a pretty critical bug. Prior to this the write statement was w(msgBuffer[:n]) for that very reason, but somehow it has been switched to w(msgBuffer). That's probably on me
